src/components/file-explorer.tsx

```
@@ -107,16 +107,19 @@ export const FileExplorer = ({ files }: FileExplorerProps) => {
-      setTimeout(() => {
+      toast.success("Copied to clipboard");
+    } catch {
+      toast.error("Failed to copy to clipboard");
+    } finally {
+      const timeout = setTimeout(() => {
         setCopied(false);
       }, 2000);
-      toast.success("Copied to clipboard");
-    } else {
-      toast.error("No file selected");
+      return () => clearTimeout(timeout);
```

Biome (2.1.2)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)